src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java

```diff
@@ -72,12 +85,44 @@
         };
     }
 
+    @Override public OutputStream decorateLogger(AbstractBuild build, OutputStream logger)
+            throws IOException, InterruptedException, RunnerAbortedException {
+
+        Map<String,String> overrides = new HashMap<String, String>();
+        for (MultiBinding<?> binding : bindings) {
+            MultiBinding.MultiEnvironment environment = binding.bind(build, null, null, null);
+            build.addAction(new SharedMultiEnvAction(environment));
```

Comment:
https://github.com/escoem/credentials-binding-plugin/blob/773fa40ab3fed6954080f596e5f8668a65c40d46/src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java#L65-L86

decorateLogger is called (only one time) before setUp